src/sentry/runner/commands/run.py

```diff
@@ -66,6 +66,21 @@ def convert(self, value, param, ctx):
 QueueSet = QueueSetType()
 
 
+def strict_offset_reset_option():
+    return click.option(
+        "--strict-offset-reset/--no-strict-offset-reset",
+        default=True,
+        help=(
+            "--strict-offset-reset, the default, means that the kafka consumer "
+            "still errors in case the offset is out of range.\n\n"
+            "--no-strict-offset-reset will use the auto offset reset even in that case. "
+            "This is useful in development, but not desirable in production since expired "
+            "offsets mean data-loss.\n\n"
+            "Most consumers that do not have this option at all default to 'Not Strict'."
```

Comment:
a little confused by this last line, by "most consumer that don't have this option at all", do you mean consumers that dont use arroyo?

Reply:
there are legacy consumers in sentry that use BatchingKafkaConsumer for which the `strict_offset_reset` kwarg is not implemented and that therefore cannot have this CLI parameter. All other consumers, the ones that use arroyo's newer streaming consumer, I've added the CLI parameter.

From what I can tell the legacy-type coonsumers never had this behavior where auto-offset-reset would be overridden with "error" in case of out of bounds offsets, but the newer ones do.